### ISSUE TITLE
Own POMDP model

### ISSUE THREAD

**Issue by friedsela** [namoshizun/PyPOMDP#1]
Very nice project. Thanks!
I wanted to check it on my own POMDP model but I get errors. It seems that just changing the file name doesn't work. Been spending some time trying to find the problem...

**Comment by namoshizun** [comment]
Thanks @friedsela !
Could you give a bit more descriptive details regarding the model you have created, and what are the error messages? I should be able to look into the issue later this weekend.

**Comment by friedsela** [comment]
For example, taking the file 1d.POMDP from http://www.pomdp.org/examples/ gives the error

    'mab_bv1': UtilityFunction.mab_bv1(min(self.model.costs), C),
TypeError: 'NoneType' object is not iterable

If I add the line 
costs: 0 0  
to the file
gives the error

  File "mtrand.pyx", line 1146, in mtrand.RandomState.choice
ValueError: probabilities do not sum to 1

**Comment by namoshizun** [comment]
Hi @friedsela, I've just pushed a batch of minor fixes for addressing some issues in the POMDP parser and the utility function creation. But that didn't really solve the issue you encountered when using on the 1d.POMDP dataset.

Basically, the current implementation does not yet querying the action reward given (s_i, s_j, o), see line 89 in [model.py](https://github.com/namoshizun/PyPOMDP/blob/master/pypomdp/models/model.py). The only supported is (s_i), i.e., reward conditioned only on the current state, which was the only case of study during the development of this package..

**Comment by friedsela** [comment]
So, if I understand correctly, if in my POMDP the reward depends only on s_i and a_i, then it should work? 
Also, as far as I understand, this is only a problem of the parser, right? For the solvers it doesn't matter?

**Comment by namoshizun** [comment]
> So, if I understand correctly, if in my POMDP the reward depends only on s_i and a_i, then it should work?
> Also, as far as I understand, this is only a problem of the parser, right? For the solvers it doesn't matter?

Yes, you are correct. The only supported POMDP reward specification is like: 
**R: my-action : current-state : * : * 100**

I couldn't remember exactly whether the parser already supports the full (s_i, s_j, o) spec... maybe it already does but it is just I chose to not consider it in the solver...

**Comment by friedsela** [comment]
Ok, great! I actually have my own implementation for POMCP, but as I need to solve many models, I wanted to see weather there is a faster implementation than my own. I also wanted to see weather point-based is faster than POMCP. In your experience, which one is faster?

**Comment by friedsela** [comment]
Thanks for your reply.

Fortunately in my model the rewards are only dependent on the action and the current state, so i'm able now to run your code. But I don't understand what's going on. Could you please explain what is actually happening and what the output means?

What I am looking for is to run a solver and then get a policy for which I can calculate the mean reward (by simulating many games).

What does '30 games played. Toal reward = 10.0' mean?
That my POMDP was solved and 30 games were played and the mean is 10?
